sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/OrcReadBenchmark.scala

```diff
@@ -0,0 +1,435 @@
+            spark.sql("SELECT sum(p), sum(id) FROM nativeOrcTable").collect()
+          }
+        }
+
+        benchmark.addCase("Read both columns - Native ORC Vectorized") { _ =>
+          spark.sql("SELECT sum(p), sum(id) FROM nativeOrcTable").collect()
+        }
+
+        benchmark.addCase("Read both columns - Hive built-in ORC") { _ =>
+          spark.sql("SELECT sum(p), sum(id) FROM hiveOrcTable").collect()
+        }
+
+        /*
+        Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16 on Mac OS X 10.13.1
+        Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
+
+        Partitioned Table:                      Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
+        ------------------------------------------------------------------------------------------------------
+        Read data column - Native ORC MR              1560 / 1562         10.1          99.2       1.0X
+        Read data column - Native ORC Vectorized       294 /  301         53.4          18.7       5.3X
+        Read data column - Hive built-in ORC          2101 / 2128          7.5         133.6       0.7X
+        Read partition column - Native ORC MR         1080 / 1087         14.6          68.7       1.4X
+        Read partition column - Native ORC Vectorized   54 /   58        289.5           3.5      28.7X
+        Read partition column - Hive built-in ORC     1315 / 1316         12.0          83.6       1.2X
+        Read both columns - Native ORC MR             1581 / 1591          9.9         100.5       1.0X
+        Read both columns - Native ORC Vectorized      329 /  339         47.8          20.9       4.7X
+        Read both columns - Hive built-in ORC         2124 / 2158          7.4         135.0       0.7X
+        */
+        benchmark.run()
+      }
+    }
+  }
+
+  def stringDictionaryScanBenchmark(values: Int): Unit = {
+    val benchmark = new Benchmark("String Dictionary", values)
+
+    withTempPath { dir =>
+      withTempTable("t1", "nativeOrcTable", "hiveOrcTable") {
+        spark.range(values).createOrReplaceTempView("t1")
+
+        prepareTable(dir, spark.sql("SELECT CAST((id % 200) + 10000 as STRING) AS c1 FROM t1"))
+
+        benchmark.addCase("Native ORC MR") { _ =>
+          withSQLConf(SQLConf.ORC_VECTORIZED_READER_ENABLED.key -> "false") {
+            spark.sql("SELECT sum(length(c1)) FROM nativeOrcTable").collect()
+          }
+        }
+
+        benchmark.addCase("Native ORC Vectorized") { _ =>
+          spark.sql("SELECT sum(length(c1)) FROM nativeOrcTable").collect()
+        }
+
+        benchmark.addCase("Hive built-in ORC") { _ =>
+          spark.sql("SELECT sum(length(c1)) FROM hiveOrcTable").collect()
+        }
+
+        /*
+        Java HotSpot(TM) 64-Bit Server VM 1.8.0_152-b16 on Mac OS X 10.13.1
+        Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz
+
+        String Dictionary:                       Best/Avg Time(ms)    Rate(M/s)   Per Row(ns)   Relative
```

Comment:
why call it `String Dictionary`? How can we make sure the data is dictionary encoded in the ORC file?